SCU (single compilation unit) build docs (#7478)
godotengine/godot-docs: contributing/development/compiling/introduction_to_the_buildsystem.rst

     $env:SCONSFLAGS="-j4"

SCU (single compilation unit) Build
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> Regular builds tend to be bottlenecked by including large numbers of headers in each compilation translation unit. Primarily to speed up development (rather than production builds), Godot offers a "single compilation unit" build (aka "Unity / Jumbo Build").

Reviewer: ```suggestion
Regular builds tend to be bottlenecked by including large numbers of headers in each compilation translation unit. Primarily to speed up development (rather than for production builds), Godot offers a "single compilation unit" build (aka "Unity / Jumbo Build").
```

For the folders accelerated by this option, multiple cpp files are compiled in each translation unit, so headers can be shared between multiple files, which can dramatically decrease build times.

> To make a SCU build, use the following scons option:

Reviewer 2: ```suggestion
To make a SCU build, use the following SCons option:
```

.. code-block:: shell

    scons scu_build={none|dev|all}

You can choose to make SCU build active for ``dev_build`` only, or for all builds.

> .. note:: When developing a PR using SCU builds, be sure to make a regular build prior to submitting the PR. This is because SCU builds by nature include headers from earlier cpps in the translation unit, therefore won't catch all the includes you will need in a regular build. The CI will catch these errors but it will usually be faster to catch them on a local build on your machine.

Reviewer: ```suggestion
.. note:: When developing a Pull Request using SCU builds, be sure to make a regular build prior to submitting the PR. This is because SCU builds by nature include headers from earlier cpps in the translation unit, therefore won't catch all the includes you will need in a regular build. The CI will catch these errors but it will usually be faster to catch them on a local build on your machine.
```

> .. note:: When developing a Pull Request using SCU builds, be sure to make a regular build prior to submitting the PR. This is because SCU builds by nature include headers from earlier cpps in the translation unit, therefore won't catch all the includes you will need in a regular build. The CI will catch these errors but it will usually be faster to catch them on a local build on your machine.

Reviewer 2: ```suggestion
.. note:: When developing a Pull Request using SCU builds, be sure to make a regular build prior to submitting the PR. This is because SCU builds by nature include headers from earlier ``.cpp`` files in the translation unit, therefore won't catch all the includes you will need in a regular build. The CI will catch these errors but it will usually be faster to catch them on a local build on your machine.
```
